p2p/discv5/net.go

```diff
@@ -580,6 +597,14 @@ loop:
 		case <-refreshDone:
 			debugLog("<-net.refreshDone")
 			refreshDone = nil
+			list := searchReqWhenRefreshDone
+			searchReqWhenRefreshDone = nil
+			go func() {
+				time.Sleep(time.Second)
```

Comment:
Why is there a delay here? If the refresh needs to wait for something else it's better to synchronise with that explicitly.